src/env.cc

```diff
@@ -933,6 +949,54 @@ char* Environment::Reallocate(char* data, size_t old_size, size_t size) {
   return new_data;
 }
 
+void AsyncRequest::Install(Environment* env, void* data, uv_async_cb target) {
+  Mutex::ScopedLock lock(mutex_);
+  env_ = env;
+  async_ = new uv_async_t;
+  async_->data = data;
+  CHECK_EQ(uv_async_init(env_->event_loop(), async_, target), 0);
+}
+
+void AsyncRequest::Uninstall() {
+  Mutex::ScopedLock lock(mutex_);
+  if (async_ != nullptr) {
+    env_->CloseHandle(async_, [this](uv_async_t* async) {
+      delete async;
+      this->async_ = nullptr;
```

Comment:
In an earlier version of this PR, this line was outside of the callback… is there a specific reason that you changed this? To me it made sense to do this, to avoid accidental double closing of the handle

Reply:
yes, I just corrected it now, as I was seeing crashes with the old code.

Reply:
@gireeshpunathil Do you know why you were seeing those crashes? That might be good to figure out either way…

Reply:
ok, I think it was my misinterpretation of some unrelated failure to be root-caused by this sequence. My reasoning was that once the `CloseHandle` is despatched, we were nullying the `async_` which could adversely affect the `uv_close` , but that does not seem to be the case, as we are passing the pointer by value and just nullifying the reference.

I ran the test several times with the original code, and do not see any problem. So I will revert this.